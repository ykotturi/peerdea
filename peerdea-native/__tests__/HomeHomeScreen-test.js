import React from 'react';
import renderer from 'react-test-renderer';

import HomeHome from '../screens/HomeHomeScreen';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);

describe('Home home Screen', () => {
  jest.useFakeTimers();

  it(`renders the home home screen without any groups`, () => {
    const mockNavigation = {
      navigate: jest.fn(),
      addListener: jest.fn().mockImplementation((event, callback) => {
        callback();
        //returning value for `navigationSubscription`
        return { remove: jest.fn() }
      }),
    };

    const tree = renderer.create(
      <HomeHome navigation={mockNavigation} />);
    expect(tree.toJSON()).toMatchSnapshot();

    var textList = tree.root.findAllByType('Text')
    
    textList.forEach((el) => {
      if (el.props.children === 'Join or create a group here!') {
        el.parent.parent.props.onClick()
        expect(mockNavigation.navigate).toHaveBeenCalledWith('GroupOptions')
      }
    })
  });
});
