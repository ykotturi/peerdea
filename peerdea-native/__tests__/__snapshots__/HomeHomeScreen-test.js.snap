// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home home Screen renders the home home screen without any groups 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Text
    style={
      Object {
        "color": "rgba(96,100,109, 1)",
        "fontSize": 25,
        "lineHeight": 40,
        "textAlign": "center",
      }
    }
  >
    You're not a member of any Peerdea groups yet!
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "color": "rgba(96,100,109, 1)",
          "fontSize": 17,
          "lineHeight": 30,
          "textAlign": "center",
          "textDecorationLine": "underline",
        }
      }
    >
      Join or create a group here!
    </Text>
  </View>
</View>
`;
